### PR TITLE
feat: Profiles

### PR DESCRIPTION
**Pull request by Shivani-1524**
added Edit user data, Search Users, Suggest Users, Like list modal, follow and unfollow feature

Files to review:
ProfilePage Folder
ModalComponent.js
ProfileThumbnail.js
SiceSection Folder
HomePage,js
postsSlice.js
modalSlice.js
store.js
TemplatePage.js

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -1,51 +1,186 @@
-.App {
-  text-align: center;
+@import url('https://fonts.googleapis.com/css2?family=Audiowide&family=Inter&display=swap');
+@import url("https://nifty-tesla-85dda2.netlify.app/Components/components.css");
+
+:root {
+  --blue-gradient: #40c9ff;
+  --violet-gradient: #e81cff;
+  --dark-bg: #1A1C1E;
+  font-size: 13px;
```
Comment (dipen2000) [inline src/App.css:8]: use rem or em here

File: src/App.css [hunk 2]
```diff
@@ -1,51 +1,186 @@
...
+}
+.icon-size-md{
+  font-size: 1.6rem;
+  text-decoration: none;
+  color: var(--white);
+}
+
+.btn.cta-btn{
+  height: min-content;
+  padding: 5px 12px;
```
Comment (dipen2000) [inline src/App.css:51]: here too

File: src/Components/CreatePost/CreatePost.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
+import React, { useState } from 'react'
+import { FontAwesomeIcon, faImage, faCircleXmark } from '../../Assets/icons/icons';
+import './createpost.css'
+import { GiphyPicker } from './GiphyPicker/GiphyPicker';
+import { ModalComponent } from '../ModalComponent/ModalComponent';
+import { useDispatch } from 'react-redux';
+import { createPost } from '../../Features/postsSlice';
+
```
Comment (Ankur9669) [inline src/Components/CreatePost/CreatePost.js:8]: use barrel exports

File: src/Components/CreatePost/GiphyPicker/GiphyPicker.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import React from 'react'
+import './giphypicker.css'
+import { useEffect, useState } from 'react'
+import { useDispatch } from 'react-redux'
+import { gifSelection } from '../../../Features/postsSlice'
+import { hideModal } from '../../../Features/modalSlice'
+
```
Comment (Ankur9669) [inline src/Components/CreatePost/GiphyPicker/GiphyPicker.js:7]: Same

File: src/Components/CreatePost/createpost.css [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+    display: block;
+    width: 100%;
+    min-height: 10rem;
+    border: none;
+    padding: 20px;
+    color: var(--white);
+}
+.post-media-row.flex-row{
+    justify-content: space-between;
+    padding: 10px 30px;
```
Comment (Ankur9669) [inline src/Components/CreatePost/createpost.css]: Use rems if possible

File: src/Components/Navbar/Navbar.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import { NavLink, Link } from "react-router-dom"
+import NavLogo from '../../Assets/images/nav-logo.png'
+import './navbar.css'
+import { faCompass, faBookmark, faPlus, FontAwesomeIcon, faRightFromBracket, faHouseChimneyCrack, faBars, faCircleUser } from '../../Assets/icons/icons'
+import { useDispatch, useSelector } from 'react-redux';
+import { showModal } from '../../Features/modalSlice';
+import { userLogout } from '../../Features/authSlice';
+
```
Comment (Ankur9669) [inline src/Components/Navbar/Navbar.js:9]: Same

**Review by dipen2000 — APPROVED** [review]
UI is very good , the features are working as expected.
Appreciation : your class names are on point.
Code looks fairly readable. Good job.

File: src/Features/postsSlice.js [hunk 7]
```diff
@@ -0,0 +1,318 @@
...
+    console.log("Comment data", { commentData, postId, commentId })
+    try {
+        const { data } = await axios.post(`/api/comments/edit/${postId}/${commentId}`, { commentData })
+        return data.posts
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+export const deleteComment = createAsyncThunk(`/${namespace}/deleteComment`, async ({ postId, commentId }, { rejectWithValue }) => {
+    console.log("daacomms", { postId, commentId })
```
Comment (Ankur9669) [inline src/Features/postsSlice.js]: Remove console.log statement

File: src/Features/postsSlice.js [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+        const { data } = await axios.post(`/api/comments/add/${postId}`, { commentData })
+        console.log("COMMENTS DATA", data)
+        return data.posts
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+
+})
+export const editComment = createAsyncThunk(`/${namespace}/editComment`, async ({ postId, commentId, commentData }, { rejectWithValue }) => {
+    console.log("Comment data", { commentData, postId, commentId })
```
Comment (Ankur9669) [inline src/Features/postsSlice.js]: Remove console.log statement

File: src/Features/postsSlice.js [hunk 9]
```diff
@@ -0,0 +1,318 @@
...
+    } catch (err) {
+        return rejectWithValue(err.response.data);
+    }
+})
+
+
+//comments
+
+export const addComment = createAsyncThunk(`/${namespace}/addComment`, async ({ postId, commentData }, { rejectWithValue }) => {
+    console.log("in add comm", { commentData })
```
Comment (Ankur9669) [inline src/Features/postsSlice.js]: Remove console.log statement

File: src/Features/postsSlice.js [hunk 10]
```diff
@@ -0,0 +1,318 @@
...
+                    state.posts.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+                    state.sortedBy = "newest"
+                    break;
+                case 'oldest':
+                    state.posts.sort((a, b) => new Date(a.createdAt) - new Date(b.createdAt))
+                    state.sortedBy = "oldest"
+                    break;
+                default:
+                    return state.posts.sort((a, b) => new Date(b.createdAt) - new Date(a.createdAt))
+            }
```
Comment (Ankur9669) [inline src/Features/postsSlice.js:148]: Appreciation: Nice use of switch statement

File: src/Features/usersSlice.js [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+    initialState: { value: initialStateValue },
+    // reducers: {
+    //     showModal: (state, action) => {
+    //         state.value.type = action.payload.type;
+    //         state.value.display = true;
+    //     },
+    //     hideModal: (state) => {
+    //         state.value = initialStateValue
+    //     }
+    // },
```
Comment (Ankur9669) [inline src/Features/usersSlice.js]: Remove commented code

File: src/Pages/AuthPages/LoginPage.js [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+                    <p className="md-title center-txt">Sign In</p>
+                    <form>
+                        <FormInput onChange={handleChange} formErrors={formErrors}
+                            props={{ labelFor: 'username', labelTitle: 'Username', value: loginData.username, placeholderText: 'Enter a Unique Username', inputType: 'text' }} />
+                        <FormInput onChange={handleChange} formErrors={formErrors}
+                            props={{ labelFor: 'password', value: loginData.password, labelTitle: 'Password', placeholderText: "Enter test", inputType: 'password' }} />
+                        {/* feature to be implemented forgotpswd */}
+                        {/* <div className="flex-between mg-t-20">
+                            <p className="orange-txt pointer">Forgot Your Password?</p>
+                        </div> */}
```
Comment (Ankur9669) [inline src/Pages/AuthPages/LoginPage.js:67]: remove commented code

**Review by Ankur9669 — APPROVED** [review]

File: src/Features/postsSlice.js (hunk 10 above)
Comment (Shivani-1524) [reply src/Features/postsSlice.js:148]: thenks

**Pull request merged**